helm/fiftyone-teams-app/values.yaml

```diff
@@ -32,6 +32,13 @@ apiSettings:
     # -- Image tag for teams-api. Defaults to the chart version.
     tag: ""
 
+  # -- Overrides the `FIFTYONE_API_URI` environment variable when set
+  # `FIFTYONE_API_URI` controls the value shown in the API Key Modal providing guidance
+  # for connecting to the FiftyOne Teams API.
+  # `FIFTYONE_API_URI` uses the value from apiSettings.dnsName if it is set, or
+  # uses the teamsAppSettings.dnsName
+  fiftyoneApiOverride: ""
```

Comment:
If this value is used to set an environment variable, why not place it in the dictionary/map of `apiSettings.env`?  If we default it to an empty string, we can still use the helm go template syntax and follow the cascasing logic to set it?

Reply:
Because I'm two out of three cases the end user doesn't set anything at all, and all of the elements here end up as env vars.

If we set it in env it will get caught in the loop below and set to ""

Reply:
Yup.  It feels weird to control the value of an environment variable outside of the env section. And that there are helm template ways to avoid it getting the value of an empty string.

Reply:
Not without parsing it once in helper (as all env vars are) and once in the template, allowing for a doubling of places for things to go awry

Reply:
Further digging suggests we'd have to rewrite how we add env vars

https://github.com/voxel51/fiftyone-teams-app-deploy/blob/main/helm%2Ffiftyone-teams-app%2Ftemplates%2Fteams-app-deployment.yaml#L38

We'd have to add some logic to loop through the list _unless_ it was the `FIFTYONE_API_URI` var.

I saw you approved it, but I thought I'd look at the alternative and now the alternative frightens me 😜

Reply:
I think that it we might be unintentionally talking past each other.  

To me, it's pretty simple to change it be something like:

```
{{- if .Values.apiSettings.fiftyoneApiOverride }}
- name: FIFTYONE_API_URI
  value: {{ .Values.apiSettings.fiftyoneApiOverride }}
{{- else if .Values.apiSettings.dnsName }}
  value: {{ printf "https://%s" .Values.apiSettings.dnsName }}
{{- else }}
  value: {{ printf "https://%s" .Values.teamsAppSettings.dnsName }}
{{- end }}
```

And Robert is your mother's brother.

Reply:
I approved because it may not be worth the extra effort.  But to me, we are creating a bit of tech debt here. But I don't want to block your progress.

Reply:
I'm confused; isn't the example you provided what I wrote in _helper?

Reply:
Oops. Haste makes waste. I just updated it.

Reply:
Right, but just below those lines in _helper it loops through all the env keys; so we would wind up with a second entry that would be later in the env list and override what we set it to.

I know, because that was my first try 😆

Reply:
Ah. Well, unless we need to set empty values, I would just in that loop check for non-empty string. If empty, don't add?

But that's why I approved. : )

I also don't like the name `override` cause they are all overrides. It's like going to the Automatic Teller Machine (ATM) Machine.

Reply:
We have values, like FIFTYONE_SERVER, that need to be set as blank.

...for... ...reasons 😆

Reply:
😿